### PR TITLE
set connection idle timeout for created ALBs

### PR DESCRIPTION
**Pull request by szuecs**
implements feature #155 set connection idle timeout for created ALBs

### CODE REVIEW

**Comment by szuecs** [comment]
@mikkeloscar if you have time to review would be nice :)

File: aws/cf_template.go [hunk 1]
```diff
@@ -2,12 +2,13 @@ package aws
...
 import (
 	"encoding/json"
+	"fmt"
 	"time"
 
 	"github.com/mweagle/go-cloudformation"
 )
 
-func generateTemplate(certs map[string]time.Time) (string, error) {
+func generateTemplate(certs map[string]time.Time, idleConnectionTimeoutSeconds uint) (string, error) {
```
[the remark this answers is not in the document]
Comment (szuecs) [reply aws/cf_template.go]: Honestly it was way too complicated for me to do and digging into undocumented Xhit, I did not wanted to do. So I decided to go with the understandable version. :)

File: aws/cf_template.go [hunk 2]
```diff
@@ -98,6 +99,10 @@ func generateTemplate(certs map[string]time.Time) (string, error) {
 		}
 	}
 	template.AddResource("LB", &cloudformation.ElasticLoadBalancingV2LoadBalancer{
+		LoadBalancerAttributes: &cloudformation.ElasticLoadBalancingV2LoadBalancerLoadBalancerAttributeList{
+			cfAttribute("idle_timeout.timeout_seconds", fmt.Sprintf("%d", idleConnectionTimeoutSeconds)),
```
[the remark this answers is not in the document]
Comment (szuecs) [reply aws/cf_template.go]: ok

File: controller.go [hunk 3]
```diff
@@ -69,6 +70,8 @@ func loadSettings() error {
 	flag.DurationVar(&healthcheckInterval, "health-check-interval", aws.DefaultHealthCheckInterval,
 		"sets the health check interval for the created target groups. The flag accepts a value "+
 			"acceptable to time.ParseDuration")
+	flag.DurationVar(&idleConnectionTimeout, "idle-connection-timeout", aws.DefaultIdleConnectionTimeout,
+		"sets the idle connection timeout if the ALB, The flag accepts a value acceptable to time.ParseDuration")
```
[the remark this answers is not in the document]
Comment (szuecs) [reply controller.go]: done

File: aws/cf_template.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (szuecs) [reply aws/cf_template.go]: done

**Comment by szuecs** [comment]
:+1:

**Comment by szuecs** [comment]
I tested it on Friday, it worked fine for new created and updated stacks. It does not update the stack itself if we have a change in this config. @mikkeloscar should we merge it or should we redo the PR to do it via template parameters?

**Comment by mikkeloscar** [comment]
:+1:

**Comment by mikkeloscar** [comment]
@szuecs I'm fine with doing the parameter cleanup later.

**Pull request merged**